app/models/group_assignment.rb

```diff
@@ -43,6 +43,22 @@ def to_label
     label
   end
 
+  def hours_since_last_submitted
+    if submitted_at.present?
+      group_offer.hours.where('created_at > ? AND volunteer_id = ?', submitted_at, volunteer)
```

Comment:
Same as above. Put that into `Hour` scopes, and split up the parts in chained wheres. The `Hour` model has the information needed ready as well...